docs/DeveloperGuide.md

```diff
@@ -1,35 +1,298 @@
+- calls on the `Storage` component to save the data of the `TaskList` component to the hard disk.
+
+### Ui component
+
+The code for this component is found in [`Ui.java`](../src/main/java/seedu/todolist/ui/Ui.java).
+
+### Logic component
+
+The main code for this component is found in [`Parser.java`](../src/main/java/seedu/todolist/logic/Parser.java).
+
+![LogicClassDiagram](images/LogicClassDiagram.png)
+
+The `Logic` component contains the `Parser`, `Command`, `ParserUtil`, and `FormatterUtil` classes, as well as
+extensions of the `Command` class like the `AddTaskCommand`, `ListTagsCommand`, and `EditEmailCommand` subclasses.
+
+When the `ToDoListManager` component passes user input to the `Parser`, a `Command` object (such as an
+`AddTaskCommand`) is created and returned to the `ToDoListManager`. The `ToDoListManager` executes the returned
+`Command` object, which will act on the `TaskList` if needed and output the result of the command through the `Ui`
+component.
+
+### TaskList component
+
+The code for this component is found in [`TaskList.java`](../src/main/java/seedu/todolist/task/TaskList.java).
+
+### Storage component
+
+The code for this component is found in [`Storage.java`](../src/main/java/seedu/todolist/storage/Storage.java).
+
+The Storage component can save the task list as TaskList objects in a .txt file format using Serialization and read it 
+back into a TaskList object.
+
+## Implementation
+
+> The lifeline of the sequence diagrams in this section should end at the destroy marker (X), but due to a limitation of PlantUML, the lifeline reaches the end of the diagram.
+
+### Delete task feature
+
+The DeleteTaskCommand extends NUS To-Do List with a delete feature for the removal of tasks from the task list.
+It is facilitated by ToDoListManager, Parser, exception, TaskList and Storage classes.
+It implements the `TaskList#deleteTask()` operation.
+
+Given below is an example usage scenario and how the DeleteTaskCommand mechanism behaves at each step.
+
+Step 1: The user launches the program for the first time. The ToDoListManager will be initialised. This in turn will
+then initialise the Parser, TaskList and Storage. Take it as there are no existing tasks read/stored by the program.
+
+Step 2: The user executes `add survey -due 20/03/2023 23:59` command to add a task for the To-Do List.
+The add command calls `TaskList#addTask()`, which causes a new Task to be added to the existing TaskList.
+
+Step 3: The user now then decides that adding this task was a mistake, and decides to remove the task from the
+To-Do List. The user does this by inputting the command `delete 1` into the terminal to delete a task in the task list.
+The command will then call the `TaskList#deleteTask()`, which removes the task at id 1 of the TaskList.
+
+The following sequence diagram shows how the delete task operation works:
+
+![DeleteTaskCommandSequence](images/DeleteTaskCommandSequence.png)
```

Comment:
I like this Sequence diagram, The information is very clear and the colors make it very understand too!